src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +430,1174 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_02()
+        {
+            var source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable disable
+    static void M1
+#nullable enable
+        ()
+    {
+        object obj = null;
+    }
+}";
+            verify(source, new[] { "M1" },
+                // (9,22): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object obj = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(9, 22));
+
+            source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable disable
+    static void M2()
+#nullable enable
+    {
+        object obj = null;
+    }
+}";
+            verify(source, new[] { "M2" },
+                // (8,22): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object obj = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(8, 22));
+
+            source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable disable
+    static void M3()
+    {
+#nullable enable
+        object obj = null;
+    }
+}";
+            verify(source, new[] { "M3" },
+                // (8,22): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object obj = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(8, 22));
+
+            source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable disable
+    static void M4()
+    {
+#nullable disable
+        object obj = null;
+    }
+#nullable enable
+}";
+            verify(source, new string[0]);
+
+            source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable enable
+    static void M5
+#nullable disable
+        ()
+    {
+        object obj = null;
+    }
+}";
+            verify(source, new[] { "M5" });
+
+            source =
+@"#pragma warning disable 219
+static class Program
+{
+    static void M6
+#nullable enable
+        ()
+#nullable disable
+    {
+        object obj = null;
+    }
+}";
+            verify(source, new[] { "M6" });
+
+            source =
+@"static class Program
+{
+    static object M7()
+#nullable enable
+        => default(object);
+}";
+            verify(source, new[] { "M7" });
+
+            source =
+@"static class Program
+{
+#nullable disable
+    static object M8() =>
+        default(
+#nullable enable
+            object);
+}";
+            verify(source, new[] { "M8" });
+
+            static void verify(string source, string[] expectedAnalyzedKeys, params DiagnosticDescription[] expectedDiagnostics)
+            {
+                var comp = CreateCompilation(source);
```

Comment:
As these (and the other similar tests in this file) depend on a specific global context, consider being explicit about it in the tests.

Reply:
> As these (and the other similar tests in this file) depend on a specific global context, consider being explicit about it in the tests.

Is the suggestion to add an explicit `.WithNullableContextOptions(null)`? If so, I'd prefer to leave it unset, since that matches most nullable tests.

---
In reply to: [546896848](https://github.com/dotnet/roslyn/pull/49953#discussion_r546896848) [](ancestors = 546896848)